nms/v1_11_R1/src/main/java/vg/civcraft/mc/namelayer/misc/v1_11_R1/ProfileModifier.java

```diff
@@ -0,0 +1,87 @@
+package vg.civcraft.mc.namelayer.misc.v1_11_R1;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.logging.Logger;
+
+import net.minecraft.server.v1_11_R1.EntityHuman;
+
+import org.bukkit.Bukkit;
+import org.bukkit.craftbukkit.v1_11_R1.CraftServer;
+import org.bukkit.craftbukkit.v1_11_R1.entity.CraftHumanEntity;
+import org.bukkit.entity.Player;
+
+import vg.civcraft.mc.namelayer.misc.ProfileInterface;
+
+import com.mojang.authlib.GameProfile;
+
+public class ProfileModifier implements ProfileInterface {
+
+	private static final Logger log = Logger.getLogger(ProfileModifier.class.getSimpleName());
+
+	@Override
+	public void setPlayerProfle(Player player, String name) {
+		String oldName = player.getName();
+		if (name.length() > 16) {
+			log.info(String.format("The player %s (%s) was kicked from the server due to his "
+							+ "name already existing but now becoming over 16 characters.",
+					name, player.getUniqueId().toString()));
+		}
+		try {
+			// start of getting the GameProfile
+			CraftHumanEntity craftHuman = (CraftHumanEntity) player;
+			EntityHuman human = craftHuman.getHandle();
+			GameProfile prof = human.getProfile();
```

Comment:
Well this is a welcome improvement; finally a first class Bukkit api call to get the GameProfile? Bravo.